src/main/java/org/opensearch/plugin/insights/core/service/categorizer/SearchQueryCounters.java

```diff
@@ -83,17 +87,17 @@ public SearchQueryCounters(MetricsRegistry metricsRegistry) {
         this.queryTypeLatencyHistogram = metricsRegistry.createHistogram(
             "search.query.type.latency.histogram",
             "Histogram for the latency per query type",
-            UNIT
+            UNIT_MILLIS
         );
         this.queryTypeCpuHistogram = metricsRegistry.createHistogram(
             "search.query.type.cpu.histogram",
             "Histogram for the cpu per query type",
-            UNIT
+            UNIT_CPU_CYCLES
         );
         this.queryTypeMemoryHistogram = metricsRegistry.createHistogram(
             "search.query.type.memory.histogram",
             "Histogram for the memory per query type",
-            UNIT
+            UNIT_BYTES
```

Comment:
I am assuming the units will not be explicitly written for each record!?

Reply:
Nope looks like we are not writing it with every record. Just adding it here for our reference or in case we use it in the future